apps/versions/models.py

```diff
@@ -252,6 +255,14 @@ def compatible_platforms(self):
             all_plats.update(amo.MOBILE_PLATFORMS)
         return all_plats
 
+    @property
+    def failed_full_review(self):
+        return any(a.activity_log.details.get('reviewtype') == 'nominated'
+                   for a in (self.all_activity_query.filter(
+                       activity_log__action__in=(
+                           amo.LOG.REJECT_VERSION.id,
+                           amo.LOG.PRELIMINARY_VERSION.id))))
```

Comment:
I'm still not happy with this. Using **logs** for anything else than displaying history feels wrong. Even more for business logic.

Please add a field on `Version`. It could be something like `rejected_date`, or `failed_review_date`. I guess it could simply be a boolean (because we already have the `Version.reviewed` field which stores the last time it was reviewed).

We could even go as far as adding two fields, `Version.full_nomination` and `Version.full_reviewed` which would be the equivalent to `Version.nomination` and `Version.reviewed`. In that case, wouldn't we have already all the needed information? It would also give us a better understanding of what's the state of the version, without going through its history.

@kmaglione @clouserw thoughts?